### PR TITLE
Suggestions Component Test & Refactoring Code

### PR DESCRIPTION
**Pull request by fakhruddinkw**
migrated the fetch call to rtk query, wrote new tests, refactored the component

test coverage:
suggestions-test

### CODE REVIEW

File: __tests__/Unit/Components/Tasks/Suggestions.test.tsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+                    handleAssignment={handleAssignment}
+                    handleClick={handleClick}
+                    assigneeName="xyz"
+                    showSuggestion={true}
+                />
+            </Provider>
+        );
+        const input = screen.getByTestId('assignee-input');
+        fireEvent.change(input, { target: { value: 'afsf' } });
+        expect(handleAssignment).toHaveBeenCalled();
```
Comment (Pratiyushkumar) [inline __tests__/Unit/Components/Tasks/Suggestions.test.tsx:33]: change the assignee name and input value, xyz and afsf does not look good

File: src/components/tasks/SuggestionBox/Suggestions.tsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        return (
+            <div className={classNames.suggestionDiv}>
+                <input
+                    data-testid="assignee-input"
+                    ref={ref}
+                    value={assigneeName}
+                    className={classNames.cardStrongFont}
+                    onChange={(e) => handleAssignment(e)}
+                    role="button"
+                    tabIndex={0}
```
Comment (Pratiyushkumar) [inline src/components/tasks/SuggestionBox/Suggestions.tsx:43]: we dont give role=button to input field, either remove it or assign some sensible role to it

**Review by Pratiyushkumar — DISMISSED** [review]

File: __tests__/Unit/Components/Tasks/Suggestions.test.tsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+import handlers from '../../../../__mocks__/handlers';
+const server = setupServer(...handlers);
+describe('Suggestions', () => {
+    beforeAll(() => server.listen());
+    afterEach(() => server.resetHandlers());
+    afterAll(() => server.close());
+    const handleClick = jest.fn();
+    const handleAssignment = jest.fn();
+    it('should return User not found', async () => {
```
Comment (sahsisunny) [inline __tests__/Unit/Components/Tasks/Suggestions.test.tsx:19]:
```suggestion
    const handleAssignment = jest.fn();
    
    it('should return User not found', async () => {
```

File: __tests__/Unit/Components/Tasks/Suggestions.test.tsx [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+import { fireEvent, screen, waitFor, render } from '@testing-library/react';
+import { renderWithRouter } from '@/test_utils/createMockRouter';
+import Suggestions from '@/components/tasks/SuggestionBox/Suggestions';
+import { Provider } from 'react-redux';
+import { store } from '@/app/store';
+import { setupServer } from 'msw/node';
+
+import usersHandler from '../../../../__mocks__/handlers/users.handler';
+
+import handlers from '../../../../__mocks__/handlers';
```
Comment (sahsisunny) [inline __tests__/Unit/Components/Tasks/Suggestions.test.tsx:11]:
```suggestion
import React from 'react';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import { Provider } from 'react-redux';
import { store } from '@/app/store';
import { setupServer } from 'msw/node';
import { renderWithRouter } from '@/test_utils/createMockRouter';

import Suggestions from '@/components/tasks/SuggestionBox/Suggestions';

import handlers from '../../../../__mocks__/handlers';
import usersHandler from '../../../../__mocks__/handlers/users.handler';
```

File: __tests__/Unit/Components/Tasks/Suggestions.test.tsx [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+import { renderWithRouter } from '@/test_utils/createMockRouter';
+import { Provider } from 'react-redux';
+import { store } from '@/app/store';
+import { setupServer } from 'msw/node';
+
+import usersHandler from '../../../../__mocks__/handlers/users.handler';
+import Suggestions from '@/components/tasks/SuggestionBox/Suggestions';
+
+import handlers from '../../../../__mocks__/handlers';
+const server = setupServer(...handlers);
```
Comment (Pratiyushkumar) [inline __tests__/Unit/Components/Tasks/Suggestions.test.tsx:12]: why spaces between import...it's not required and also there should be space after line 11.

**Review by Pratiyushkumar — APPROVED** [review]

**Review by sahsisunny — APPROVED** [review]

**Review by vinit717 — APPROVED** [review]

**Pull request merged**